pandas/core/dtypes/base.py

```diff
@@ -391,6 +391,32 @@ def _can_hold_na(self) -> bool:
         return True
 
 
+class StorageExtensionDtype(ExtensionDtype):
+    """ExtensionDtype that may be backed by more than one implementation."""
+
+    name: str
+    na_value = libmissing.NA
+    _metadata = ("storage",)
+
+    def __init__(self, storage=None) -> None:
+        self.storage = storage
+
+    def __repr__(self):
+        return f"{self.name}[{self.storage}]"
+
+    def __str__(self):
+        return self.name
+
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, self.type) and other == self.name:
```

Comment:
Do you remember why you added this line here? (it's different as the line in `__eq__` you moved here from StringDtype, and this only works "by chance" for StringDtype, as its `.type` is `str`)

So I think you can actually keep the `isinstance(other, str)` as it was originally in StringDtype, as this is meant to check for strings (regardless of the type of the dtype).


(from https://github.com/pandas-dev/pandas/pull/47034#discussion_r880938565)

Reply:
Hmm yeah I am not sure why this changed. This was just supposed to be a copy-paste essentially to a new subclass and maybe make it more generic maybe since `StringDtype.type == str`, but clearly I misunderstood why this was `str` in the first place.